tracer/src/Datadog.Trace/Agent/Transports/ApiWebRequest.cs

```diff
@@ -62,21 +63,44 @@ public async Task<IApiResponse> PostAsJsonAsync(IEvent events, JsonSerializer se
 
             using (var requestStream = await _request.GetRequestStreamAsync().ConfigureAwait(false))
             {
-                using (var writer = new JsonTextWriter(new StreamWriter(requestStream)))
+                static Task WriteStream(Stream stream, JsonSerializer serializer, object events)
                 {
-                    serializer.Serialize(writer, events);
-                    await writer.FlushAsync();
-                    try
+                    var streamWriter = new StreamWriter(stream, System.Text.Encoding.UTF8, 1024, true);
+                    using (var writer = new JsonTextWriter(streamWriter))
                     {
-                        var httpWebResponse = (HttpWebResponse)await _request.GetResponseAsync().ConfigureAwait(false);
-                        return new ApiWebResponse(httpWebResponse);
+                        serializer.Serialize(writer, events);
+                        return writer.FlushAsync();
                     }
-                    catch (WebException exception)
-                        when (exception.Status == WebExceptionStatus.ProtocolError && exception.Response != null)
+                }
+
+                await WriteStream(requestStream, serializer, events);
```

Comment:
I spotted some more `ConfigureAwait(false)` 😅 
```suggestion
                await WriteStream(requestStream, serializer, events).ConfigureAwait(false);
```